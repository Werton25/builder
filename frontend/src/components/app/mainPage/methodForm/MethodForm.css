.spacing {
    margin-top: 10px !important;
    margin-left: 10px !important;
}