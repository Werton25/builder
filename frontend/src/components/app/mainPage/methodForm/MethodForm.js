import React from 'react';
import PropTypes from 'prop-types';
import RequestTabs from "./tabs/RequestTabs";
import ResponseTabs from "./tabs/ResponseTabs";
import {postRequest} from "../../../../actions/requests";
import {connect} from 'react-redux';
import './MethodForm.css';
import {getCurrentRequest} from "../../../../selectors/requests";
import {getCurrentUri} from "../../../../selectors/router";

export class BaseMethodForm extends React.Component {
  componentWillMount() {
    this.setState({response: ''});
  }

  onSubmit = (e) => {
    e.preventDefault();
    const {currentUri, currentRequest, postRequest} = this.props;
    postRequest(currentUri, currentRequest).then((response) => this.setState({response}));
  };

  render() {
    return (
      <form className='form-vertical' onSubmit={this.onSubmit}>
        <div className='row'>
          <div className='col-sm-6'>
            <RequestTabs/>
          </div>
          <div className='col-sm-6'>
            <ResponseTabs response={this.state.response}/>
          </div>
        </div>
        <div className='pull-right spacing'>
          <button className='btn btn-primary' type='submit'>Отправить</button>
        </div>
      </form>
    );
  }
}

BaseMethodForm.propTypes = {
  currentUri: PropTypes.string.isRequired,
  currentRequest: PropTypes.object.isRequired,
  postRequest: PropTypes.func.isRequired
};

const mapStateToProps = (state) => ({
  currentUri: getCurrentUri(state),
  currentRequest: getCurrentRequest(state)
});

const mapDispatchToProps = {
  postRequest
};

export default connect(mapStateToProps, mapDispatchToProps)(BaseMethodForm);
