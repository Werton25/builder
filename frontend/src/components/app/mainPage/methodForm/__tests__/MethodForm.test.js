import React from "react";
import renderer from 'react-test-renderer';
import {Provider} from "react-redux";
import configureStore from "redux-mock-store";
import MethodForm from "../MethodForm";

jest.mock('connected-react-router');
jest.mock('../tabs/RequestTabs');
jest.mock('../tabs/ResponseTabs');
jest.mock('../../../../../selectors/router', () => ({getCurrentUri: () => '/service/method'}));
jest.mock('../../../../../selectors/requests', () => ({
  getCurrentRequest: () => ({
    id: '1',
    name: 'test',
    value: '{}'
  })
}));
const store = configureStore()({});

describe('MethodForm tests', () => {
  it('MethodForm renders correctly', () => {
    const tree = renderer
      .create(
        <Provider store={store}>
          <MethodForm/>
        </Provider>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

});

