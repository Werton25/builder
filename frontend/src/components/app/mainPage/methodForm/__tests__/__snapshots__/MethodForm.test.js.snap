// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MethodForm tests MethodForm renders correctly 1`] = `
<form
  className="form-vertical"
  onSubmit={[Function]}
>
  <div
    className="row"
  >
    <div
      className="col-sm-6"
    />
    <div
      className="col-sm-6"
    />
  </div>
  <div
    className="pull-right spacing"
  >
    <button
      className="btn btn-primary"
      type="submit"
    >
      Отправить
    </button>
  </div>
</form>
`;
